### DISCUSSION TITLE
Stopnet loss got high and not reducing dataset change

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#314] [Archived]
<USER>>>> vcjobacc<br/> <DATE> [December 4, 2020, 1:45pm]<br/> <br/> Hello everyone!<br/> <br/> Since I've got limited hardware and I want to speed up training as much<br/> as I can, I first pretrain tts on short utterances, and then I change<br/> the dataset (same speaker) to the one with longer utterances (and<br/> smaller batch size). *slash* <br/> The thing is, the stopnet got much worse and seems like it's not going<br/> to be improved. I use separate stopnet. Here what plots look like: *slash* <br/> <br/> Is there any way it can be fixed? Can I somehow increase LR of stopnet<br/> or freeze the Tacotron's weight for a while?<br/> <br/> Thank you!!<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/stopnet-loss-got-high-and-not-reducing-dataset-change]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [December 4, 2020, 2:18pm]<br/> <br/> I'd check longer samples if they have silence at the end and how long<br/> this offset is. There might be a difference b/w short and long samples<br/> and this affects stopnet after you switch.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> vcjobacc<br/> <DATE> [December 4, 2020, 8:04pm]<br/> <br/> They have no silence on the end, indeed! I've checked all records, they<br/> have like 0.1 - 0.2 sec silence on the end. *slash* <br/> According to the attention plots, attention is also struggling. If<br/> listen to the eval and test audios, they usually end with a silence,<br/> like multiple seconds. *slash* <br/> <br/> *slash* <br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [December 4, 2020, 4:03pm]<br/> <br/> How is the performance when testing it in novel sentences? I think it is<br/> overfitting looking at the graphs, but I might be wrong. But the<br/> alignment graph also does not seem that it is capturing a lot of<br/> information.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> vcjobacc<br/> <DATE> [December 4, 2020, 4:20pm]<br/> <br/> *slash* <br/> it depends. Some are good, some are bad. The thing is, I used multiple<br/> sentences at once in dataset. On the small dataset there were primarily<br/> one-sentence utterances, and now it's mostly multiple-sentence<br/> utterances. It does impact the performance. Anything I can do about it?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> vcjobacc<br/> <DATE> [December 4, 2020, 4:21pm]<br/> <br/> all test files are multisentence. But it skips last sentences from some<br/> of them.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> vcjobacc<br/> <DATE> [December 4, 2020, 8:10pm]<br/> <br/> May it help to train stopnet<br/> NOT separately?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [December 4, 2020, 8:35pm]<br/> <br/> Okay, if stopnet is the only problem, you can try not training it when<br/> you restore the model that has been trained on smaller utterances.<br/> Generally, silences are not very good for attention. If what you want is<br/> for the TTS to sound natural in between sentences, pairs (two sentences<br/> at most per wav) are enough for the TTS to pick up on silence modelling.<br/> I imagine my workflow in your position would be to first evaluate the<br/> quality of the training set (sound and content, e.g. if transcriptions<br/> match what is spoken). After that, do what you did and train on the<br/> short sentences. And after this is robust enough, train on a dataset<br/> that has at most two sentences per wav file and the silence between them<br/> is not longer than 0.100 secs. Also, take care that all other silences<br/> are short (while the speaker is talking).<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> vcjobacc<br/> <DATE> [December 5, 2020, 10:19am]<br/> <br/> Thank you! *slash* <br/> But how do I freeze weight for the stopnet and train the rest of the<br/> tacotron only?<br/> <br/> [Archived Post]<br/>
